client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java

```diff
@@ -344,6 +344,11 @@ private PartitionReader createReaderWithRetry(
           if (isExcluded(location)) {
             throw new CelebornIOException("Fetch data from excluded worker! " + location);
           }
+          // should switch the location at a high level
+          if (fetchChunkRetryCnt == 0 && attemptNumber % 2 == 1 && location.hasPeer()) {
+            location = location.getPeer();
```

Comment:
Do we need to consider the case of `pbStreamHandler == null` ?

Reply:
This situation does not need to be considered because WorkerPartitionReader has been made compatible